plugins/modules/ipadnszone.py

```diff
@@ -0,0 +1,453 @@
+}
+
+DOCUMENTATION = """
+module: ipadnszone
+short description: Manage FreeIPA dnszone
+description: Manage FreeIPA dnszone
+options:
+  ipaadmin_principal:
+    description: The admin principal
+    default: admin
+  ipaadmin_password:
+    description: The admin password
+    required: false
+
+  forwarders:
+    description: The list of global DNS forwarders.
+    required: false
+    options:
+      ip_address:
+        description: The forwarder nameserver IP address list (IPv4 and IPv6).
+        required: true
+      port:
+        description: The port to forward requests to.
+        required: false
+  forward_policy:
+    description:
+      Global forwarding policy. Set to "none" to disable any configured
+      global forwarders.
+    required: false
+    choices: ['only', 'first', 'none']
+  allow_sync_ptr:
+    description:
+      Allow synchronization of forward (A, AAAA) and reverse (PTR) records.
+    required: false
+    type: bool
+  state:
+    description: State to ensure
+    default: present
+    choices: ["present", "absent", "enabled", "disabled"]
+  name_server:
+    description: Authoritative nameserver domain name
+    required: false
+    type: str
+  admin_email:
+    description: Administrator e-mail address
+    required: false
+    type: str
+  update_policy: BIND update policy
+    description: Allow dynamic updates
+    required: false
+    type: str
+  dnssec:
+    description: Allow inline DNSSEC signing of records in the zone
+    required: false
+    type: bool
+  allow_transfer:
+    description: List of IP addresses or networks which are allowed to transfer the zone
+    required: false
+    type: bool
+  allow_query:
+    description: List of IP addresses or networks which are allowed to issue queries
+    required: false
+    type: bool
+  serial:
+    description: SOA record serial number
+    required: false
+    type: int
+  refresh:
+    description: SOA record refresh time
+    required: false
+    type: int
+  retry:
+    description: SOA record retry time
+    required: false
+    type: int
+  expire:
+    description: SOA record expire time
+    required: false
+    type: int
+  minimum:
+    description: How long should negative responses be cached
+    required: false
+    type: int
+  ttl:
+    description: Time to live for records at zone apex
+    required: false
+    type: int
+  default_ttl:
+    description: Time to live for records without explicit TTL definition
+    required: false
+    type: int
+  nsec3param_rec:
+    description: NSEC3PARAM record for zone in format: hash_algorithm flags iterations salt.
+    required: false
+    type: str
+  skip_overlap_check:
+    description: Force DNS zone creation even if it will overlap with an existing zone
+    required: false
+    type: bool
+  skip_nameserver_check:
+    description: Force DNS zone creation even if nameserver is not resolvable
+    required: false
+    type: bool
+"""  # noqa: E501
+
+EXAMPLES = """
+---
+- ipadnszone:
```

Comment:
For each of this example we usually add a comment `#` to explain what the example should do if used.